Shared/Supporting Files/Views/SearchResultView.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SearchResultView: View {
+    /// All samples retrieved from the Samples directory.
+    let samples: [Sample]
+    
+    /// The search query in the search bar.
+    let query: String
+    
+    /// The search result to display in the various sections.
+    @State private var searchResult = SearchResult(nameMatches: [], descriptionMatches: [], tagMatches: [])
+    
+    var body: some View {
+        List {
+            if !searchResult.nameMatches.isEmpty {
+                Section(header: Text("Name Results")) {
+                    ForEach(searchResult.nameMatches, id: \.name) { sample in
+                        SampleRow(sample: sample, boldedText: query)
+                    }
+                }
+            }
+            if !searchResult.descriptionMatches.isEmpty {
+                Section(header: Text("Description Results")) {
+                    ForEach(searchResult.descriptionMatches, id: \.name) { sample in
+                        SampleRow(sample: sample, boldedText: query)
+                    }
+                }
+            }
+            if !searchResult.tagMatches.isEmpty {
+                Section(header: Text("Tags Results")) {
```

Comment:
This `Section` initializer is deprecated.

```suggestion
                Section("Tags Results") {
```

Although consider using "Tag Results" as it reads better and the result matches a single tag.

```suggestion
                Section("Tag Results") {
```